ui/src/app/pages/admin/disks-table.spec.tsx

```diff
@@ -0,0 +1,200 @@
+
+import { screen } from '@testing-library/dom';
+import {
+  fireEvent,
+  render,
+  waitFor,
+  waitForElementToBeRemoved,
+  within,
+} from '@testing-library/react';
+import * as swaggerClients from 'app/services/swagger-fetch-clients';
+import moment from 'moment';
+
+import { DisksTable } from './disks-table';
+
+const convertDate = (originalDate: String) => {
+  const date = originalDate.substring(0, 10);
+  const hour =
+    parseInt(originalDate.substring(11, 13), 10) + moment().utcOffset() / 60;
+  const minutes = originalDate.substring(14, 16);
+  return `${date} ${hour}:${minutes}`;
+};
+
+const getEnvironmentType = (isGceRuntime: boolean, appType: AppType) => {
+  return isGceRuntime ? 'Jupyter' : fp.capitalize(appType.toString());
+};
+
+const mockJupyterDisk: Disk = {
+  size: 1000,
+  diskType: DiskType.Standard,
+  isGceRuntime: true,
+  name: 'mock-disk1',
+  blockSize: 1,
+  status: DiskStatus.Ready,
+  appType: null,
+  creator: '"evrii@fake-research-aou.org"',
+  createdDate: '2023-05-22T18:55:10.108838Z',
+};
+
+const mockCromwellDisk: Disk = {
+  size: 1000,
+  diskType: DiskType.Standard,
+  isGceRuntime: false,
+  name: 'mock-disk2',
+  blockSize: 1,
+  status: DiskStatus.Ready,
+  appType: AppType.CROMWELL,
+  creator: '"evrii@fake-research-aou.org"',
+  createdDate: '2023-05-22T18:55:10.108838Z',
+};
+
+const mockRStudioDisk: Disk = {
+  size: 1000,
+  diskType: DiskType.Standard,
+  isGceRuntime: false,
+  name: 'mock-disk3',
+  blockSize: 1,
+  status: DiskStatus.Ready,
+  appType: AppType.RSTUDIO,
+  creator: '"evrii@fake-research-aou.org"',
+  createdDate: '2023-05-22T18:55:10.108838Z',
+};
+
+const mockDisks: ListDisksResponse = [
+  mockJupyterDisk,
+  mockCromwellDisk,
+  mockRStudioDisk,
+];
+
+let mockdisksAdminApi;
+
+beforeEach(() => {
+  jest.mock('app/services/swagger-fetch-clients');
+  mockdisksAdminApi = jest.spyOn(swaggerClients, 'disksAdminApi');
+});
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+const setup = (mockOverrides) => {
+  mockdisksAdminApi.mockImplementation(() => ({
+    ...mockOverrides,
+  }));
+};
+
+test('loads and displays table', async () => {
+  setup({ listDisksInWorkspace: () => Promise.resolve(mockDisks) });
+  render(<DisksTable sourceWorkspaceNamespace='123' />);
+  await waitForElementToBeRemoved(() =>
+    screen.getByTitle('disks loading spinner')
+  );
+
+  mockDisks.forEach((disk) => {
+    const row = screen.getByText(disk.name).closest('tr');
+    const rowScope = within(row);
+    expect(rowScope.getByText(disk.name)).toBeInTheDocument();
+    expect(rowScope.getByText(disk.creator)).toBeInTheDocument();
+    expect(
+      rowScope.getByText(convertDate(disk.createdDate))
+    ).toBeInTheDocument();
+    expect(rowScope.getByText(disk.status)).toBeInTheDocument();
+    expect(
+      rowScope.getByText(getEnvironmentType(disk.isGceRuntime, disk.appType))
+    ).toBeInTheDocument();
+    expect(rowScope.getByText(disk.size)).toBeInTheDocument();
+    expect(
```

Comment:
We are using <div role=’button> instead of <button> for most of our buttons in the app. The div tag does not support the disabled attribute, otherwise, we could check for disabled here instead.
https://github.com/rafgraph/react-interactive#disabled-boolean